job/server/src/main/java/alluxio/master/job/JobMasterClientRestServiceHandler.java

```diff
@@ -127,16 +128,23 @@ public JobInfo call() throws Exception {
 
   /**
    * Lists all the jobs in the history.
+   * @param status the target status of jobs
    *
    * @return the response of the names of all the jobs
    */
   @GET
   @Path(ServiceConstants.LIST)
-  public Response list() {
+  public Response list(@QueryParam("status") final String status) {
     return RestUtils.call(new RestUtils.RestCallable<List<Long>>() {
       @Override
       public List<Long> call() throws Exception {
-        return mJobMaster.list(ListAllPOptions.getDefaultInstance());
+        if (status != null) {
+          return mJobMaster.list(ListAllPOptions.newBuilder()
+              .addStatus(Status.valueOf(status))
```

Comment:
the api actually supports filtering jobs by multiple statuses, do we want to support that? 
users might want both completed and failed jobs in one call. It is up to you though. 

also the api supports filtering by name as well, you may find it useful to add that. we can get all the replicate jobs or evict jobs that are completed, for example.

Reply:
This is a great idea, will support it.